include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -879,7 +881,7 @@ __pattern_partition_copy(_ExecutionPolicy&& __exec, _Iterator1 __first, _Iterato
         __par_backend_hetero::zip(
             __par_backend_hetero::make_iter_mode<__par_backend_hetero::access_mode::write>(__result1),
             __par_backend_hetero::make_iter_mode<__par_backend_hetero::access_mode::write>(__result2)),
-        __create_mask_op, __copy_by_mask_op);
+        _AlgoType{}, __create_mask_op, __copy_by_mask_op);
```

Comment:
I don't really like passing the default constructed version of `_AlgoType` and discarding it, but it is of course better than specifying all the template arguments.

Putting `_AlgoType` first in the template argument list is an option, and there are other instances of doing this in oneDPL (`unseq_backend::__scan` and `__brick_fill`), but I'm not sure if it is the right call.